### DISCUSSION TITLE
Module/Idea Suggestion List.

### DISCUSSION THREAD

**Discussion by Xeddius** [GitSquared/edex-ui#481] [Ideas]
**Perceived idea awesomeness**
<details><summary>Wow, you don't have this yet???</summary>
Web Browser Module,
Background Module,
Media Module/s,
Security/Lock Module.
</details>
<details><summary>Would be super cool.</summary>
Temperature Module,
Volume Control Module.
VNC Module,
Custom Text/Matrix Module.
</details>
<details><summary>I'm not sure a lot of people would use it, but I'd love to have it.</summary>
Self Destruct/Panic button/Red Alert Module.
</details>
<details><summary>Kind of a gadget, honestly, but why not?</summary>
Rainbow/Party mode Module.
</details>

**Perceived idea difficulty**
- [0] Quit your job first.
- [3] Would need some work, but nothing impossible.
- [7] Shouldn't be too hard to make.
- [1] Like, *I* could do it.

---
**Simple/-ish ideas:**
- Web Browser Module. (Requires some work, but still might be possible.)
- Background Module. (changes background to image, slideshow, or video/animation.)
- Media Module/s. (Show image/s, video, slideshows, music/audio, etc.)
- ~~Volume Control Module.~~ (Not feasable)
- Rainbow/Party mode. (Changes colors of theme over time. Not super useful, just really fun!)
- Security/Lock Module. (Lock Screen, alarm for wrong password, eDEX-UI only, non-system lock.)
- "[noVNC](https://github.com/novnc/noVNC)" module for connecting to another device. (a phone or another computer.) *(novnc is based on javascript and html, porting _should_ be pretty simple.)
- Text Module. (Print lines from a text file, Monitor a log file, or be set to rain code like in the matrix, should be pretty easy to make as the startup sequence already prints scrolling text.)
---
**Complex ideas/Stretch Goals:**
- LCARS Theme/Module. (Not likely to be super useful, but it'd be really neat, I'll likely attempt this in the future.)
---
**In Progress/Planned Modules:**
- [x] Space/Memory Module.
- [x] External Module Support.
- [x] Temperature Module.
- [x] Disk Activity Module.
---
**Completed Modules:**
- *placeholder*
---
**Mockup/Screenshots/Concepts**

Browser Module Concept: ![edex-main-browser-interface](https://user-images.githubusercontent.com/6382895/56373617-598c4400-61cf-11e9-9906-6e3eabab23a5.png)
Text Module Concept: ![animated](https://user-images.githubusercontent.com/6382895/56390410-f8c63100-61f9-11e9-8f80-22a182ff3f23.gif)

---

(UPDATE!)
Someone created a video player for Electron using Node.js, I'm sure it can be useful for helping to create a media module in edex-ui! It's called [Playback](https://github.com/mafintosh/playback)

This app could become a pretty practical UI replacement, with some of these suggested modules.
 I would definitely love to run this as the primary GUI on a minimal Arch Linux install! (probably running in i3 or another simple/lightweight window manager.)
I'm working on learning Javascript. I might be able to make one or a few of these modules sometime in the future. I will update this post with more ideas and possibly submit code as I continue to research and learn more about Electron + Javascript.

**Comment by antares667367** [comment]
+1 for the LCARS module ^^

**Reply by csigristgit** [reply]
+47 for the LCARS module ;)

**Reply by ghost** [reply]
Here's a package that would be useful if anyone wanted to undertake that.
https://github.com/Aricwithana/LCARS-SDK

**Reply by ghost** [reply]
Looks like [CBS claims to own the copyright on LCARS](https://en.wikipedia.org/wiki/LCARS#Legal), so it would probably endanger the project to include it.

**Reply by GitSquared** [reply]
> Looks like [CBS claims to own the copyright on LCARS](https://en.wikipedia.org/wiki/LCARS#Legal), so it would probably endanger the project to include it.

I wonder if we're breaking any TRON copyright... at least I never heard from Disney so far.
